R/fread.R

```diff
@@ -35,7 +35,7 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="")
     if (!is.character(text)) stop("'text=' is type ", typeof(text), " but must be character.")
     if (!length(text)) return(data.table())
     if (length(text) > 1L) {
-      cat(text, file=(tmpFile<-tempfile(tmpdir=tmpdir)), sep="\n")  # avoid paste0() which could create a new very long single string in R's memory
+      writeLines(text, tmpFile<-tempfile(tmpdir=tmpdir))  # avoid paste0() which could create a new very long single string in R's memory
```

Comment:
We could also use `useBytes=TRUE` here to facilitate writing UTF-8 strings on Windows:

https://stackoverflow.com/q/10675360/3576984

Reply:
@MichaelChirico This isn't something I've tested, but maybe `fwrite(as.data.table(text), tmpFile<-tempfile(tmpdir = tmpdir), col.names = FALSE)` is also worth exploring. I'm basing this assumption on the speed that I observed [here](https://github.com/Rdatatable/data.table/issues/4572#issue-644031582).

Reply:
In general I think it may warrant some benchmarking. Actually I have run into an issue recently where `cat()` is very slow for long input. In my case I was `cat`-ing a whole file by character (e.g. `cat(strsplit(readChar(f, file.size(f)), NULL)[[1]], sep = '', file = f)` is quite slow)